### DISCUSSION TITLE
Nifti Images with STIR

### DISCUSSION THREAD

**Discussion by Kiarashdnsh** [UCL/STIR#1487] [Q&A]
Hello!

I'm trying to reconstruct PET images in nifti format using the SIRF VM. SIRF is the Synergistic Image Reconstruction Framework that uses the STIR module in image reconstruction. As I understand it, only images with the interfile .hv and .hs formats are usable with this module, and no Nifti or Dicom files are readable. Am I correct? I also changed the nifti format to h33, i33, and v formats using the XMedcon application in an attempt to load converted interfile formats to my code.
However, the code I use
```python
image_data = pet.ImageData('path/to/file/m000-31013_AV451_S_E.v')
``` 
renders this error:
```python
ERROR: Error opening file /media/sf_DATABASE/m000-m000-31013_AV451_S_E.v
``` 
and that
```python
the reconstruction engine output may provide more information
``` 
I get the same error message when using h33/i33 as well.

If you could help me load up nifti or dicom files with this module, I would much appreciate it. Thanks!

**Comment by evgueni-ovtchinnikov** [comment]
`pet.ImageData` expects `*.hv`-file name as the argument, not `*.v`-file.

**Reply by Kiarashdnsh** [reply]
Exactly, that's the problem i'm having. My original files are in Nifti format. I am looking for a method to load up those.

**Comment by evgueni-ovtchinnikov** [comment]
actually, as I have just checked, `pet.ImageData` can read `*.nii` file as well

**Reply by Kiarashdnsh** [reply]
How do you do that? As I try to load up nifti files, I encounter this error:
```python
ERROR: Error opening file /media/sf_PET-MR/Test Dataset/31000_AV45_S_E.nii
error: ??? "'Error opening file /media/sf_PET-MR/Test Dataset/31000_AV45_S_E.nii\\n' exception caught at line 398 of /home/sirfuser/devel/buildVM/sources/SIRF/src/xSTIR/cSTIR/cstir.cpp; the reconstruction engine output may provide more information"
``` 
when I use this code block:
```python
import sirf.STIR as pet
image = pet.ImageData('/media/sf_PET-MR/Test Dataset/31000_AV45_S_E.nii')
```

**Comment by evgueni-ovtchinnikov** [comment]
```
sirfuser@vagrant:~/Laptop$ ls *.nii
NAC_image.nii
sirfuser@vagrant:~/Laptop$ python3
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sirf.STIR as pet
>>> image = pet.ImageData('NAC_image.nii')

INFO: Reading NAC_image.nii via ITK non-DICOM IO

WARNING: Unable to determine patient position. Internally this will generally be handled by assuming HFS
>>> print(image.shape)
(127, 285, 285)
>>> print(image.norm())
17.781774520874023
```
Make sure you are using the latest versions of SIRF and STIR (I have SIRF 3.8.0 and STIR rel_6.2.0).

**Reply by Kiarashdnsh** [reply]
I'm using SIRF 3.5. I'll update to the latest version and will try again.
